File: src/main/java/com/kedzie/vbox/api/IMachineStateChangedEvent.java
package com.kedzie.vbox.api;

import com.kedzie.vbox.api.jaxb.MachineState;
import com.kedzie.vbox.soap.Cacheable;
import com.kedzie.vbox.soap.KSOAP;

public interface IMachineStateChangedEvent extends IMachineEvent {
	@Cacheable @KSOAP(prefix="IMachineStateChangedEvent") 	public MachineState getState();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IMachineStateChangedEvent.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.api.jaxb.MachineState
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap(prefix="IMachineStateChangedEvent")
interface IMachineStateChangedEvent : IMachineEvent {

    @Cacheable("state")
    suspend fun getState(): MachineState
}
